### ISSUE TITLE
Enhancement: Support direct instruction encoding (From capstone)

### ISSUE THREAD

**Issue by cynecx** [keystone-engine/keystone#193]
I think I would be nice to have direct instruction encoding supported. So it is actually possible to use both capstone and keystone together. With capstone one could decode a specific instruction, the user then could modify for example an operand, afterwards with keystone it would be possible to use a `cs_insn` and you would get the encoded code.

So instead of using `MCAsmParser` we could "_just_" use `MCCodeEmitter::encodeInstruction` directly. Obviously a conversion between `cs_insn` -> `MCInst` should happen before, this is probably the tricky part.

**Comment by aquynh** [comment]
yes this was requested by many people. we also had some discussions about it at https://github.com/keystone-engine/keystone/issues/93

**Comment by mrexodia** [comment]
If you want to do this without having to dig around the keystone codebase, just create a small class that converts a `cs_insn` structure to text. I need a parser tailored to my needs as well and the only way is to parse the instruction myself and then convert it to a format keystone accepts. It's not pretty, but can be done in quite an efficient way without having to change keystone.
